parachain/pallets/control/src/lib.rs

```diff
@@ -311,25 +389,49 @@ pub mod pallet {
 
 		pub fn convert_location(
 			mut location: MultiLocation,
-		) -> Result<(H256, Option<ParaId>, MultiLocation), DispatchError> {
+		) -> Result<(H256, ParaId, MultiLocation), DispatchError> {
 			// Normalize all locations relative to the relay chain.
 			let relay_location = T::RelayLocation::get();
 			location
 				.reanchor(&relay_location, T::UniversalLocation::get())
 				.map_err(|_| Error::<T>::LocationReanchorFailed)?;
 
-			// Only allow Parachain as origin location
-			let para_id = match location {
-				MultiLocation { parents: 0, interior: X1(Parachain(index)) } =>
-					Some((index).into()),
+			let para_id = match location.interior.first() {
+				Some(Parachain(index)) => Some((*index).into()),
 				_ => None,
-			};
+			}
+			.ok_or(Error::<T>::LocationToParaIdConversionFailed)?;
 
 			// Hash the location to produce an agent id
 			let agent_id = T::AgentHashedDescription::convert_location(&location)
 				.ok_or(Error::<T>::LocationToAgentIdConversionFailed)?;
 
 			Ok((agent_id, para_id, location))
 		}
+
+		pub fn account_id() -> T::AccountId {
+			PalletId(*b"snow/ctl").into_account_truncating()
+		}
+
+		pub fn reserve_deposit(
+			payer: T::AccountId,
+			reserve_deposit: Option<BalanceOf<T>>,
+		) -> DispatchResult {
+			if reserve_deposit.is_some() {
```

Comment:
I think we can simplify this code by just passing a fee of `0` instead of Option. Then `T::Token::transfer` may just handle the zero fee case gracefully.